tests/impl/uJIT-tests-C/suite/test_lua_yield.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * This is a part of uJIT's testing suite.
+ * Copyright (C) 2015-2019 IPONWEB Ltd. See Copyright Notice in COPYRIGHT
+ *
+ * This test is based on the reproducer provided by Arseny Vakhrushev
+ * within https://github.com/iponweb/luavela/issues/12.
+ */
+
+#include "test_common_lua.h"
+
+/* To test passing values via stack on yield: */
+#define YIELD_STATUS "yielded"
+
+static int yield(lua_State *L)
+{
+	assert_stack_size(L, 0);
+	lua_pushliteral(L, YIELD_STATUS);
+	return lua_yield(L, lua_gettop(L));
+}
+
+static void assert_yielded(lua_State *L, const char *status)
+{
+	assert_int_equal(lua_status(L), LUA_YIELD);
+	assert_stack_size(L, 1);
+	test_string(L, -1, status);
+	lua_pop(L, 1);
+}
+
+static void assert_global(lua_State *L, const char *name, const char *value)
```

Comment:
Minor: I guess this one can be moved to `test_common_lua.h`, since it looks like a generic check. Feel free to ignore :)

Reply:
Good point, fixed. Thanks!